Examples/Io/Root/src/RootTrackSummaryWriter.cpp

```diff
@@ -221,8 +217,14 @@ ProcessCode RootTrackSummaryWriter::finalize() {
 ProcessCode RootTrackSummaryWriter::writeT(const AlgorithmContext& ctx,
                                            const ConstTrackContainer& tracks) {
   // Read additional input collections
-  const auto& particles = m_inputParticles(ctx);
-  const auto& trackParticleMatching = m_inputTrackParticleMatching(ctx);
+  const SimParticleContainer* particles = nullptr;
+  const TrackParticleMatching* trackParticleMatching = nullptr;
+
+  if (m_inputParticles.isInitialized() &&
+      m_inputTrackParticleMatching.isInitialized()) {
+    particles = &m_inputParticles(ctx);
+    trackParticleMatching = &m_inputTrackParticleMatching(ctx);
+  }
```

Comment:
alternatively you could use references and if `m_inputParticles` is not preset for example let it point to a static empty one.

something like

```suggestion
  static SimParticleContainer emptyParticles;

  const SimParticleContainer particles = m_inputParticles.isInitialized() ? m_inputParticles(ctx) : emptyParticles;
```

Reply:
@andiwand your diff would now do a copy, you'd need to add the `&`.

Reply:
Thanks I corrected the diff